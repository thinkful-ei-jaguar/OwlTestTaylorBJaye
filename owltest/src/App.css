* {
    box-sizing: border-box;
}

.App {
    display: flex;
    flex-direction: row;
}

.stageBox{
    display: flex;
    background-color: gray;
    width: 1000px;
    height: 800px;
}
