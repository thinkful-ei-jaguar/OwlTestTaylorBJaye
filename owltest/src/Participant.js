import React from 'react';
import './participant.css';

export default function Participant (props) {
    let p;
    if(props.onStage)
        p = 'onStage';
    else if(props.inSession)
        p = 'inSession';
    else
        p = 'Left Session';
    return (
        <div>
            <div className="session-participant">
                <img src = {props.avatar} alt={props.name} className="user-avatar"></img>
            </div>
            <div className="participant-info">
                <h6 className="participant-name">{props.name}</h6>
                <div className="participant-status participant-status__green">
                    {p}
                </div>
            </div>
        </div>
    )
}