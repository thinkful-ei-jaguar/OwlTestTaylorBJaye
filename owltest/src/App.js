import React, {Component} from 'react';
import { render } from 'react-dom';
import Participant from './Participant';
import Stage from './Stage';
import './App.css';

class App extends Component {
  static defaultProps = {
    participants: {}
  };

  render() {
    const participants = this.props
    return (
      <main className='App'>

        <div className="participants-list">
          {participants.participantStore.map(person => (
            <Participant
              key={person.id}
              name={person.name}
              avatar={person.avatar}
              inSession={person.inSession}
              onStage={person.onStage}
            />
          ))}
        </div>

        <div className='stageBox'>
        {participants.participantStore.map(person => (
            <Stage
              key={person.id}
              name={person.name}
              avatar={person.avatar}
              inSession={person.inSession}
              onStage={person.onStage}
            />
          ))}
        </div>

      </main>
    );
  }
}

export default App;