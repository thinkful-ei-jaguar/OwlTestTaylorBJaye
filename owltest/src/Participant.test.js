import React from 'react';
import ReactDOM from 'react-dom';
import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';
import App from './App';
import Participant from './Participant';

describe('Participant Component', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Participant
      id= {123}
      name='Kitten'
      avatar= 'https://placekitten.com/200/287'
      inSession= {true}
      onStage= {true}
    />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders the Participant UI as expected', () => {
    const tree = renderer
      .create(<Participant
        id= {123}
        name='Kitten'
        avatar= 'https://placekitten.com/200/287'
        inSession= {true}
        onStage= {true}
      />)
      .toJSON();
    expect(tree).toMatchSnapshot();  
  });
  /*
  it('should be a leap year if divisible by 4, not 100', () => {
    let input = <Participant
        id= {123}
        name='Kitten'
        avatar= 'https://placekitten.com/200/287'
        inSession= {true}
        onStage= {true}
        />;
    const expectedOutput = 'onStage';
    const actualOutput = Participant(input);
    expect(actualOutput).toBe(expectedOutput);
  });
  */  
});
