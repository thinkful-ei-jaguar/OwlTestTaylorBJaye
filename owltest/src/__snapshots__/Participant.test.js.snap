// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Participant Component renders the UI as expected 1`] = `
<div>
  <div
    className="session-participant"
  >
    <img
      alt="Kitten"
      className="user-avatar"
      src="https://placekitten.com/200/287"
    />
  </div>
  <div
    className="participant-info"
  >
    <h6
      className="participant-name"
    >
      Kitten
    </h6>
    <div
      className="participant-status participant-status__green"
    >
      onStage
    </div>
  </div>
</div>
`;
